R/modify_pd.R

```diff
@@ -9,33 +9,27 @@ NULL
 
 #' @describeIn update_indention Inserts indetion based on round brackets.
 indent_round <- function(pd, indent_by) {
-  indention_needed <- needs_indention(pd, token = "'('")
+  opening <- which(pd$token == "'('")
+  indention_needed <- needs_indention(pd, token = "'('", opening[1])
   if (indention_needed) {
-    opening <- which(pd$token == "'('")
     start <- opening + 1
     stop <- nrow(pd) - 1
-    if (start > stop) return(pd)
-
-    pd <- pd %>%
-      mutate(indent = indent + ifelse(seq_len(nrow(pd)) %in% start:stop,
-                                      indent_by, 0))
+    pd$indent <- pd$indent +
+      ifelse(between(seq_len(nrow(pd)), start, stop), indent_by, 0)
   }
 
   pd %>%
     set_unindention_child(token = "')'", unindent_by = indent_by)
 }
 #' @rdname update_indention
 indent_curly <- function(pd, indent_by) {
-  indention_needed <- needs_indention(pd, token = "'{'")
+  opening <- which(pd$token == "'{'")
+  indention_needed <- needs_indention(pd, token = "'{'", opening[1])
```

Comment:
Can we define a function `get_indent_indices()` instead that returns a (possibly empty) integer vector of positions that need to have indention added?

```r
indent_indices <- get_indent_indices(pd, ...)
if (length(indent_indices) > 0L) pd$indent[indent_indices] <- pd$indent[indent_indices] + 2L
```

Also, the benchmarks suggest that this doesn't buy us anything.